File: particle.py
# particle.py
import numpy as np
from constants import ELECTRON_CHARGE, ELECTRON_MASS, MU_ZERO, EPSILON_ZERO, GRAVITY_ACCELERATION

class Particle:
    def __init__(self, charge, mass, initial_velocity=[0,0,0], initial_position=[0,0,0], name=__name__):
        self.name = name
        self.charge = charge
        self.mass = mass
        self.velocity = np.array(initial_velocity, dtype=float)
        self.position = np.array(initial_position, dtype=float)
        self.trajectory = []  # List to store trajectory points

        # Constants for the particle
        self.epsilon_0 = EPSILON_ZERO  # Permittivity of free space
        self.mu_0 = MU_ZERO  # Permeability of free space
        self.gravity_acceleration = GRAVITY_ACCELERATION  # Acceleration due to gravity

File: Partcile_simulator/Particle.cpp
// Particle.cpp

#include "Particle.h"

Particle::Particle(double charge, double mass, const std::array<double, 3>& initial_velocity, const std::array<double, 3>& initial_position, const std::string& name)
    : charge(charge), mass(mass), velocity(initial_velocity), position(initial_position), name(name) {
    // Other constructor initialization...
}

double Particle::getCharge() const {
    return charge;
}

double Particle::getMass() const {
    return mass;
}

std::string Particle::getName() const {
    return name;
}

// Other member function implementations...
